fenix/app/src/main/java/org/mozilla/fenix/gleanplumb/NimbusMessagingStorage.kt

```diff
@@ -53,9 +53,29 @@ class NimbusMessagingStorage(
         get() = attributeProvider?.getCustomAttributes(context) ?: JSONObject()
 
     /**
-     * Returns a list of available messages descending sorted by their priority.
+     * Returns a list of currently available messages descending sorted by their priority.
+     * This list of messages will not include any expired, pressed or dismissed messages.
      */
-    suspend fun getMessages(): List<Message> {
+    suspend fun getCurrentMessages(): List<Message> {
+        return getMessages(false)
+    }
+
+    /**
+     * Returns a list of all available messages descending sorted by their priority.
+     * This list of messages will include expired, pressed or dismissed messages.
+     */
+    suspend fun getAllMessages(): List<Message> {
+        return getMessages(true)
+    }
+
+    /**
+     * Returns a list of messages descending sorted by their priority.
+     *
+     * @param noFilter, when set to true, no filter is applied when retrieving messages.
+     */
+    private suspend fun getMessages(
+        noFilter: Boolean = false,
```

Comment:
nit: generally I think we should avoid negative Boolean names, having just 'filter' or 'shouldFilter' and flipping the logic may be cleaner?